### DISCUSSION TITLE
codechecker check segmentation fault with snap installed v.6.15.1

### DISCUSSION THREAD

**Discussion by charles-randall** [Ericsson/codechecker#3235] [General]
Per the documentation, I did a snap install and got,
```

$ codechecker version
CodeChecker analyzer version:
---------------------------------------
Kind                 | Version         
---------------------------------------
Base package version | 6.15.1          
Package build date   | 2021-02-08T09:15
Git commit ID (hash) |                 
Git tag information  |                 
---------------------------------------

CodeChecker web version:
------------------------------------------------
Kind                          | Version         
------------------------------------------------
Base package version          | 6.15.1          
Package build date            | 2021-02-08T09:15
Git commit ID (hash)          |                 
Git tag information           |                 
Server supported API (Thrift) | 6.39            
Client API (Thrift)           | 6.39            
------------------------------------------------
```

Trivial Makefile,
```
$ cat Makefile 
testfile: testfile.c
	gcc -Wall -Wextra -o testfile testfile.c
```
Coredump,
```
$ codechecker check -b make
[INFO 2021-03-20 18:07] - Starting build ...
Segmentation fault (core dumped)
[INFO 2021-03-20 18:07] - Build failed.

$ file core
core: ELF 64-bit LSB core file, x86-64, version 1 (SYSV), SVR4-style, from '/bin/sh -c make', real uid: 1000, effective uid: 1000, real gid: 1000, effective gid: 1000, execfn: '/bin/sh', platform: 'x86_64'

```

My system,
```
$ hostnamectl  | tail -3
  Operating System: Ubuntu 20.04.2 LTS
            Kernel: Linux 5.4.0-67-generic
      Architecture: x86-64
```

Suggestions?

**Comment by csordasmarton** [comment]
@charles-randall Sorry for the late response. I think the problem will be an architecture problem. Our snap package was built on an `Ubuntu 18.04.5 LTS` / `Linux 4.15.0-140-generic` / `x86-64` machine for the default architecture which is `amd64`.

I checked it and I saw that snap supports packages for multiple architectures (https://snapcraft.io/docs/architectures). So we should investigate where we can build our snap packages for multiple architectures.

Unfortunately we can't use the `build.snapcraft.io` page which is recommended by snap because it can only create snap packages from the master branch but for bugfix releases we create other branches.

[Launchpad](https://launchpad.net/) can be an option to build packages (under the hood the build.snapcraft.io uses this too) but I am not familiar with it yet and it takes more time to figure out how it works exactly.
